docs/import/score_bounds_map.js

```diff
@@ -105,21 +114,39 @@ class ScoreBoundsMap {
    * have not yet been set for the current data. Bounds cannot be set earlier
    * because {@link google.maps.Map} does not like having its bounds set when it
    * is not visible.
+   *
+   * If there is no polygon and {@link this.stateBoundsPromise} is not yet set,
+   * will set bounds asynchronously when it is, if user has not already zoomed.
+   * @return {?Promise<void>} Promise that completes when bounds are set (or
+   *     Promise to calculate bounds is complete but user zoomed map in
+   *     meantime), or null if bounds-setting was done synchronously or not
+   *     needed.
    */
   onShow() {
     if (!this.needsBoundsFit) {
-      return;
+      return null;
     }
     this.needsBoundsFit = false;
+    // Fit map around existing polygon.
+    const bounds = new google.maps.LatLngBounds();
     if (this.polygon) {
-      // Fit map around existing polygon.
-      const bounds = new google.maps.LatLngBounds();
       this.polygon.getPath().forEach((latlng) => bounds.extend(latlng));
       applyMinimumBounds(bounds, this.map);
+      return null;
     } else {
-      // TODO(janakr): center around states for disaster.
-      this.map.setCenter(defaultMapCenter);
-      this.map.setZoom(defaultZoomLevel);
+      let zoomChangedByUser = false;
+      google.maps.event.addListenerOnce(
+          this.map, 'zoom_changed', () => zoomChangedByUser = true);
+      return this.stateBoundsPromise.then((resolvedBounds) => {
+        if (zoomChangedByUser) {
+          return;
+        }
+        const coordinates = resolvedBounds.coordinates[0];
+        bounds.extend(latLngLiteralFromArray(coordinates[0]));
```

Comment:
can you document the significance of the 0 and 2 indices?

Reply:
extracted some variables, hopefully the names help.